src/Pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useCallback, ChangeEvent } from 'react';
+import { connect } from 'react-redux';
+import { actions } from '../../Store';
+import { Switch, Route, useLocation, useHistory } from 'react-router-dom';
+import { Header, MovieCard, MovieDetails, Navigation, Movie } from '../../Components';
+import css from './MainPage.less';
+
+function MainPage(
+  {
+    loading,
+    error,
+    movies,
+    addMovie,
+    updateMovie,
+    deleteMovie,
+    fetchMovies,
+    setFilteringOption,
+    setSortingOption,
+  }: {
+    loading: boolean;
+    error: null | Error;
+    movies: Movie[];
+    addMovie: (movie: Movie) => void;
+    updateMovie: (movie: Movie) => void;
+    deleteMovie: (movieId: number) => void;
+    fetchMovies: (searchString?: string) => void;
+    setFilteringOption: (filteringOption: string) => void;
+    setSortingOption: (sortingOption: string) => void;
+  },
+) {
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(
+    () => {
+      if (location.pathname === '/films' || !movies.length) {
+        fetchMovies();
+      }
+      if (location.pathname === '/search') {
+        const queryParam = location.search.match(new RegExp('[?&]searchString=([^&]+).*$'));
```

Comment:
It might be better to use here `useQuery` as well

Also, you probably don't need to use those condition check here as you can just get the query param and for `/films` it would be empty

Reply:
fixed